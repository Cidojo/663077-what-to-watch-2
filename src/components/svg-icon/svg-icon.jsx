import * as React from 'react';
import PropTypes from 'prop-types';

const SvgIcon = (props) => {
  const {name, viewBox, width, height} = props;

  return (
    <svg viewBox={viewBox} width={width} height={height}>
      <use xlinkHref={`#${name}`}></use>
    </svg>
  );
};

SvgIcon.propTypes = {
  name: PropTypes.string,
  viewBox: PropTypes.string,
  width: PropTypes.number,
  height: PropTypes.number
};

SvgIcon.defaultProps = {
  name: ``,
  viewBox: `0 0 0 0`,
  width: 0,
  height: 0
};

export {SvgIcon};
