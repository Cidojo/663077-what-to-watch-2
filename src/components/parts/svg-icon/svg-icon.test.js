import * as React from 'react';
import {SvgIcon} from './svg-icon.jsx';
import * as renderer from 'react-test-renderer';

const props = {
  name: ``,
  viewBox: `0 0 0 0`,
  width: 0,
  height: 0
};

it(`should render SvgIcon component without errors`, () => {
  const tree = renderer.create(<SvgIcon {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
