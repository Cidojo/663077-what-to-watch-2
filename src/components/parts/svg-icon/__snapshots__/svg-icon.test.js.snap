// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SvgIcon component without errors 1`] = `
<svg
  height={0}
  viewBox="0 0 0 0"
  width={0}
>
  <use
    xlinkHref="#"
  />
</svg>
`;
